File: src/utility/Menu.java
package utility;

import java.util.ArrayList;
import java.util.Scanner;

public class Menu {
    public String value;
    public String name;
    public ArrayList< String > options;

    public Menu( String name, ArrayList< String > options ) {
        this.name = name;
        this.options = options;
    }

    public void display( ) {
        System.out.println( name );
        for ( int i = 1; i < options.size( ) + 1; i++ ) {
            System.out.println( "(" + i + ") " + options.get( i - 1 ) );
        }
        System.out.print( ">>> " );
        Scanner scn = new Scanner( System.in );
        value = scn.nextLine( );
    }

    public int getIntValue( ) {
        try {
            return Integer.parseInt( value );
        } catch ( Exception valueNotAnInteger ) {
            valueNotAnInteger.printStackTrace( );
            return -1;
        }
    }

    public String getStringValue( ) {
        return value;
    }
}

File: Game/Menu.cs
using System;
using System.Collections.Generic;

namespace Game
{
    public class Menu
    {
        public string value;
        public string name;
        public List<string> options;

        public Menu(string name, List<string> options)
        {
            this.name = name;
            this.options = options;
        }

        public void display()
        {
            Console.WriteLine(name);
            for (int i = 1; i < options.Count + 1; i++)
            {
                Console.WriteLine("(" + i + ") " + options[i - 1]);
            }
            Console.Write(">>> ");
            value = Console.ReadLine();
        }

        public int getIntValue()
        {
            return int.Parse(value);
        }

        public string getStringValue()
        {
            return value;
        }
    }
}